src/shared/install.c

```diff
@@ -740,6 +740,9 @@ static int find_symlinks_fd(
                         _cleanup_free_ char *p = NULL;
                         int nfd, q;
 
+                        if (endswith(de->d_name, ".d"))
+                                continue;
+
```

Comment:
i don't get this one? if this is right, then we don't need to iterate through dirs here at all?

Reply:
Yeah, this seems very iffy. I don't understand why the recursive call is being done here. The symlinks we care about are not nested more than one level... So I think this code might be just doing unnecessary work.

Reply:
I think that current recursive approach is elegant but it may be doing unnecessary work. Clearly, it is recursing into drop-in directories and there are no symlinks of interest there. So idea is to short-cut this and recurse into directories like *.wants or *.requires but not into .d directories.

Reply:
i think the fact we operated recrusively here is a historic remnant, and is something we should drop. I remeber i found it nice to be recursive here, but it doesn't really make much sense. Let's drop it.

Reply:
But will the iterative solution be much better or shorter? I mean we still need to do a bunch of work in symlink case (i.e. no change there) and for all other directories then drop-in dirs we want to process them.